Tests/AblyChatTests/IntegrationTests.swift

```diff
@@ -82,25 +96,99 @@ struct IntegrationTests {
+        let rxPresenceLeaveTxEvent = try #require(await rxPresenceSubscription.first { _ in true })
+        #expect(rxPresenceLeaveTxEvent.action == .leave)
+        #expect(rxPresenceLeaveTxEvent.data?.userCustomData?["randomData"]?.value as? String == "randomValue")
+
+        // (21) Send `.enter` presence event with custom data on our client and check that we receive it on the subscription
+        try await txRoom.presence.enter(data: .init(userCustomData: ["randomData": .string("randomValue")]))
+        let rxPresenceEnterRxEvent = try #require(await rxPresenceSubscription.first { _ in true })
+        #expect(rxPresenceEnterRxEvent.action == .enter)
+        #expect(rxPresenceEnterRxEvent.data?.userCustomData?["randomData"]?.value as? String == "randomValue")
+
+        // (22) Send `.update` presence event with custom data on our client and check that we receive it on the subscription
+        try await txRoom.presence.update(data: .init(userCustomData: ["randomData": .string("randomValue")]))
+        let rxPresenceUpdateRxEvent = try #require(await rxPresenceSubscription.first { _ in true })
+        #expect(rxPresenceUpdateRxEvent.action == .update)
+        #expect(rxPresenceUpdateRxEvent.data?.userCustomData?["randomData"]?.value as? String == "randomValue")
+
+        // (23) Send `.leave` presence event with custom data on our client and check that we receive it on the subscription
+        try await txRoom.presence.leave(data: .init(userCustomData: ["randomData": .string("randomValue")]))
+        let rxPresenceLeaveRxEvent = try #require(await rxPresenceSubscription.first { _ in true })
+        #expect(rxPresenceLeaveRxEvent.action == .leave)
+        #expect(rxPresenceLeaveRxEvent.data?.userCustomData?["randomData"]?.value as? String == "randomValue")
+
         // MARK: - Detach
 
-        // (12) Detach the room
+        // (24) Detach the room
         try await rxRoom.detach()
 
-        // (13) Check that we received a DETACHED status change as a result of detaching the room
+        // (25) Check that we received a DETACHED status change as a result of detaching the room
         _ = try #require(await rxRoomStatusSubscription.first { $0.current == .detached })
         #expect(await rxRoom.status == .detached)
 
         // MARK: - Release
 
-        // (14) Release the room
+        // (26) Release the room
         try await rxClient.rooms.release(roomID: roomID)
 
-        // (15) Check that we received a RELEASED status change as a result of releasing the room
+        // (27) Check that we received a RELEASED status change as a result of releasing the room
         _ = try #require(await rxRoomStatusSubscription.first { $0.current == .released })
         #expect(await rxRoom.status == .released)
 
-        // (16) Fetch the room we just released and check it’s a new object
+        // (28) Fetch the room we just released and check it’s a new object
```

Comment:
I feel like it should be different tests instead of a markup inside one giant test.